frame/support/src/traits/metadata.rs

```diff
@@ -31,6 +31,16 @@ pub trait PalletInfo {
 	fn name<P: 'static>() -> Option<&'static str>;
 }
 
+/// Provides information about the pallet setup in the runtime.
+///
+/// Access the information provided by [`PalletInfo`] for a specific pallet.
+pub trait PalletInfoAccess {
```

Comment:
Is `PalletInfo` already taken? `Access` is weird.

Reply:
`PalletInfo` is the trait giving the info for all pallets https://substrate.dev/rustdocs/v3.0.0/frame_support/traits/trait.PalletInfo.html
It is implemented by an associated type on frame-system https://substrate.dev/rustdocs/v3.0.0/frame_system/pallet/trait.Config.html#associatedtype.PalletInfo

Ideally `PalletInfo` could be renamed `PalletsInfo` as it contains info for all pallets, and thus we could use `PalletInfo` here.

Anyway those 2 traits are very related, `PalletInfoAccess` is implementing accessor to the info of the pallet. info of the pallet are retrieved from `PalletInfo` which contains info for all pallets (but which is cumbersome to use directly).

Reply:
> Ideally PalletInfo could be renamed PalletsInfo as it contains info for all pallets, and thus we could use PalletInfo here.

Totally agree with this, maybe even in this PR? the naming is weird, but the functionality is great, working with the old `PalletInfo` was not simple.

Reply:
At the same time I prefer not to break, as `PalletInfoAccess` is also an OK name to me.
Maybe we can write it somewhere for when we want to break frame stuff.

Reply:
sure, lets stay backwards compat.